src/libraries/System.Private.CoreLib/src/System/TimeSpan.cs

```diff
@@ -347,149 +349,126 @@ public static TimeSpan FromMilliseconds(double value)
 
-        public TimeSpan Negate()
-        {
-            if (Ticks == MinValue.Ticks)
-                throw new OverflowException(SR.Overflow_NegateTwosCompNum);
-            return new TimeSpan(-_ticks);
-        }
+        public TimeSpan Negate() => -this;
 
-        public static TimeSpan FromSeconds(double value)
-        {
-            return Interval(value, TicksPerSecond);
-        }
+        public static TimeSpan FromSeconds(double value) => Interval(value, TicksPerSecond);
 
-        public TimeSpan Subtract(TimeSpan ts)
-        {
-            long result = _ticks - ts._ticks;
-            // Overflow if signs of operands was different and result's
-            // sign was opposite from the first argument's sign.
-            // >> 63 gives the sign bit (either 64 1's or 64 0's).
-            if ((_ticks >> 63 != ts._ticks >> 63) && (_ticks >> 63 != result >> 63))
-                throw new OverflowException(SR.Overflow_TimeSpanTooLong);
-            return new TimeSpan(result);
-        }
+        public TimeSpan Subtract(TimeSpan ts) => this - ts;
 
         public TimeSpan Multiply(double factor) => this * factor;
 
         public TimeSpan Divide(double divisor) => this / divisor;
 
         public double Divide(TimeSpan ts) => this / ts;
 
-        public static TimeSpan FromTicks(long value)
-        {
-            return new TimeSpan(value);
-        }
+        public static TimeSpan FromTicks(long value) => new TimeSpan(value);
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         internal static long TimeToTicks(int hour, int minute, int second)
         {
             // totalSeconds is bounded by 2^31 * 2^12 + 2^31 * 2^8 + 2^31,
             // which is less than 2^44, meaning we won't overflow totalSeconds.
-            long totalSeconds = (long)hour * 3600 + (long)minute * 60 + (long)second;
-            if (totalSeconds > MaxSeconds || totalSeconds < MinSeconds)
+            long totalSeconds = (hour * SecondsPerHour)
+                              + (minute * SecondsPerMinute)
+                              + second;
+
+            if ((totalSeconds > MaxSeconds) || (totalSeconds < MinSeconds))
+            {
                 ThrowHelper.ThrowArgumentOutOfRange_TimeSpanTooLong();
+            }
             return totalSeconds * TicksPerSecond;
         }
 
         // See System.Globalization.TimeSpanParse and System.Globalization.TimeSpanFormat
         #region ParseAndFormat
-        private static void ValidateStyles(TimeSpanStyles style, string parameterName)
+        private static void ValidateStyles(TimeSpanStyles style)
         {
-            if (style != TimeSpanStyles.None && style != TimeSpanStyles.AssumeNegative)
-                throw new ArgumentException(SR.Argument_InvalidTimeSpanStyles, parameterName);
+            if ((style != TimeSpanStyles.None) && (style != TimeSpanStyles.AssumeNegative))
```

Comment:
>if ((style != TimeSpanStyles.None) && (style != TimeSpanStyles.AssumeNegative)) [](http://example.com/codeflow?start=12&length=79)

nit:

if (style is not TimeSpanStyles.None and not TimeSpanStyles.AssumeNegative)